ocs_sample_library_preview/Assets.py

```diff
@@ -0,0 +1,523 @@
+
+        result = Asset.fromJson(response.json())
+        return result
+
+    def getAssets(self, namespace_id: str, query: str = '', skip: int = 0, count: int = 100,
+                  ) -> List[Asset]:
+        """
+        Returns a list of assets
+        :param namespace_id: The namespace identifier
+        :param query: An optional query string to search for matching assets.
+        :param skip: An optional parameter representing the zero-based offset of the first asset to
+        retrieve. If not specified, a default value of 0 is used.
+        :param count: An optional parameter, between 1 and 1000 (inclusive), that represents the
+        maximum number of retrieved assets. If not specified, the default is 100.
+        """
+        if namespace_id is None:
+            raise TypeError
+
+        response = self.__base_client.request('get', self.__assets_path.format(
+            namespace_id=namespace_id), params={'skip': skip, 'count': count, 'query': query})
+        self.__base_client.checkResponse(response, f'Failed to get assets.')
+
+        results = []
+        for i in response.json():
+            results.append(Asset.fromJson(i))
+        return results
+
+    def createAsset(self, namespace_id: str, asset: Asset) -> Asset:
+        """
+        Create a new asset with a specified Id
+        :param namespace_id: The namespace identifier
+        :param asset: An asset object
+        """
+        if namespace_id is None:
+            raise TypeError
+        if asset is None or not isinstance(asset, Asset):
+            raise TypeError
+
+        response = self.__base_client.request('post', self.__asset_path.format(
+            namespace_id=namespace_id, asset_id=asset.Id), data=asset.toJson())
+        self.__base_client.checkResponse(
+            response, f'Failed to create asset, {asset.Id}.')
+
+        result = Asset.fromJson(response.json())
+        return result
+
+    def createAssets(self, namespace_id: str, assets: List[Asset]) -> List[Asset]:
+        """
+        Creates multiple assets in a single call
+        :param namespace_id: The namespace identifier
+        :param assets: An list of asset objects
+        """
+        if namespace_id is None:
+            raise TypeError
+        if assets is None or not isinstance(assets, List[Asset]):
+            raise TypeError
+
+        dictionary = []
+        for value in assets:
+            dictionary.append(value.toDictionary())
+        response = self.__base_client.request('post', self.__assets_path.format(
+            namespace_id=namespace_id), data=json.dumps(dictionary))
+        self.__base_client.checkResponse(response, f'Failed to create assets.')
+
+        results = []
+        for i in response.json():
+            results.append(Asset.fromJson(i))
+        return results
+
+    def createOrUpdateAsset(self, namespace_id: str, asset: Asset) -> Asset:
+        """
+        Create or update an asset with a specified Id
```

Comment:
same comment as line 69

Reply:
Same answer: this is copied from the description [here](https://ocs-docs.osisoft.com/Content_Portal/Documentation/Assets/asset-type.html#create-assettype). It basically means that the Id is required to be specified inside the asset for this endpoint, though it's not required when updating existing assets.